### PR TITLE
feat(#2148): Move `OyFake` into `Objectionary`

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
1. Move `OyFake` into `Objectionary` (`Objectionary.Fake`)
2. Rename `OyLocalTest` -> `OyHomeTest`

Related to #2148

---

## PR-Codex overview
This PR replaces `OyFake` with `Objectionary.Fake` in the tests of the `eo-maven-plugin` module. It also adds a new constructor to `Objectionary` interface to create a fake implementation for testing purposes.

### Detailed summary
- Replaces `OyFake` with `Objectionary.Fake` in the tests of the `eo-maven-plugin` module.
- Adds a new constructor to `Objectionary` interface to create a fake implementation for testing purposes.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Review by Graur — APPROVED** [review]
@volodya-lombrozo Good one
@yegor256 Please have a look

File: eo-maven-plugin/src/main/java/org/eolang/maven/objectionary/Objectionary.java [hunk 1]
```diff
@@ -55,4 +58,88 @@ public interface Objectionary {
...
+            this.getter = gett;
+            this.container = cont;
+        }
+
+        @Override
+        public Input get(final String name) {
+            try {
+                return this.getter.apply(name);
+            } catch (final Exception ex) {
+                Logger.debug(
```
Comment (yegor256) [inline eo-maven-plugin/src/main/java/org/eolang/maven/objectionary/Objectionary.java:121]: @volodya-lombrozo this in general is a bad practice: https://stackoverflow.com/questions/6639963/why-is-log-and-throw-considered-an-anti-pattern

**Comment by volodya-lombrozo** [comment]
@yegor256 Thank you, for the suggestion. I've fixed it. Could you take a look again, please?

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**